spyder/plugins/pylint/plugin.py

```diff
@@ -34,7 +34,7 @@ class Pylint(SpyderDockablePlugin):
     WIDGET_CLASS = PylintWidget
     CONF_SECTION = NAME
     CONF_WIDGET_CLASS = PylintConfigPage
-    REQUIRES = [Plugins.Editor]
+    REQUIRES = [Plugins.Editor, Plugins.MainMenu]

```

Comment:
Techincally the Pylint plugin could run a file without an editor (with the select file option of the combobox) so this should probably become also `OPTIONAL`